src/sources/kubernetes_logs/parser/cri.rs

```diff
@@ -173,6 +129,67 @@ impl FunctionTransform for Cri {
     }
 }
 
+struct ParsedLog<'a> {
+    timestamp: &'a [u8],
+    stream: &'a [u8],
+    multiline_tag: &'a [u8],
+    message: &'a [u8],
+}
+
+#[allow(clippy::trivially_copy_pass_by_ref)]
+#[inline]
+const fn is_delimiter(c: &u8) -> bool {
+    *c == b' '
+}
+
+/// Parses a CRI log line.
+///
+/// Equivalent to regex: `(?-u)^(?P<timestamp>.*) (?P<stream>(stdout|stderr)) (?P<multiline_tag>(P|F)) (?P<message>.*)(?P<new_line_tag>\n?)$`
+#[inline]
+fn parse_log_line(line: &[u8]) -> Option<ParsedLog> {
```

Comment:
I don't think we want to be `expect()`ing in this function, as we'd want it to still return a `None` to hit the `emit!(ParserMatchError { value: &s[..] });` arm.

Reply:
Thank you for review! I will fix this part